Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm

```diff
@@ -590,6 +592,19 @@ - (void)inspectorViewController:(WKInspectorViewController *)inspectorViewContro
         completionHandler(nullString());
 }
 
+void WebInspectorUIProxy::platformPickColorFromScreen(CompletionHandler<void(const std::optional<WebCore::Color>&)>&& completionHandler)
+{
+    NSColorSampler *sampler = [[NSColorSampler alloc] init];
```

Comment:
Is this file ARC? If not this should probably be put into a `RetainPtr` with `adoptNS` or manually released?

Reply:
You can check `leaks <pid-of-web-process>` after using the color picker a few times to see if an `NSColorSampler` leaks.

Reply:
This was indeed leaking! `adoptNS` into a `RetainPtr` fixes this leak by allowing the initial allocation to be balanced with a release at the end of the function scope. The object is still retained until the callback is called, which is managed by the `NSColorSampler` API.